src/Sarif/FileRegionsCache.cs

```diff
@@ -164,6 +164,11 @@ public Region ConstructMultilineContextSnippet(Region inputRegion, Uri uri, stri
             // Generating full inputRegion to prevent issues.
             Region originalRegion = this.PopulateTextRegionProperties(inputRegion, uri, populateSnippet: true, fileText);
 
+            if (originalRegion.CharLength >= BIGSNIPPETLENGTH)
```

Comment:
>= [](http://example.com/codeflow?start=43&length=1)

So 512 is considered big?

Reply:
+1. This should be >

Reply:
why? if the size of the original region is as big as our bigsnippet maximum, why would attempt to generate it? what am i missing?

Reply:
I was verifying that the max length is 511, since the >= operator excludes 512. This seems to be an off-by-one considering the statement that snippets are "restricted to 512 bytes in size"